Null pointer when authenticating as guest

Attempting to authenticate as guest using `AuthenticationContext.guest()` results in an exception 
```
Caused by: java.lang.NullPointerException
	at com.hierynomus.smbj.connection.SMBSessionBuilder.deriveKey(SMBSessionBuilder.java:324)
	at com.hierynomus.smbj.connection.SMBSessionBuilder.deriveKeys(SMBSessionBuilder.java:290)
	at com.hierynomus.smbj.connection.SMBSessionBuilder.setupSession(SMBSessionBuilder.java:174)
	at com.hierynomus.smbj.connection.SMBSessionBuilder.setupSession(SMBSessionBuilder.java:152)
	at com.hierynomus.smbj.connection.SMBSessionBuilder.establish(SMBSessionBuilder.java:119)
	at com.hierynomus.smbj.connection.Connection.authenticate(Connection.java:197)
```

The server is running `SMB_3_1_1` (Windows Server 2022)

--- a/src/main/java/com/hierynomus/smbj/connection/SMBSessionBuilder.java
+++ b/src/main/java/com/hierynomus/smbj/connection/SMBSessionBuilder.java
@@ -163,7 +163,7 @@ public class SMBSessionBuilder {
 
             SessionContext context = session.getSessionContext();
             processAuthenticationToken(ctx, response.getSecurityBuffer());
-            if (!ctx.authContext.isAnonymous() && !ctx.authContext.isGuest()) {
+            if (ctx.sessionKey != null) {
                 context.setSessionKey(new SecretKeySpec(ctx.sessionKey, HMAC_SHA256_ALGORITHM));
             }
             if (dialect == SMB2Dialect.SMB_3_1_1) {

